src/components/UserDetailPage/UserDetailPage.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import { Author } from '../../models/Author';
+import { Book } from '../../models/Book';
+import { Category } from '../../models/Category';
+import BookCard from '../BookCard/BookCard';
+
+function UserDetailPage() {
+
+  const [books, setBooks] = useState<Book[]>([]);
+
+  useEffect(() => {
+    fetch('https://localhost:8000/my/favorites')
+      .then(response => response.json())
+      .then(data => {
+        const res = data.data.map((book: Book) => {
+          const authors = book.authors.map((author: Author) => `${author.lastName}`);
+          const categories = book.categories.map((category: Category) => `${category.categoryName}`);
+          return { ...book, authors, categories };
+        });
+        setBooks(res);
+      })
+      .catch(error => console.error(error));
+  }, []);
+
+  useEffect(() => {
+    fetch('https://localhost:8000/my/borrowed')
+      .then(response => response.json())
+      .then(data => {
+        const res = data.data.map((book: Book) => {
+          const authors = book.authors.map((author: Author) => `${author.lastName}`);
+          const categories = book.categories.map((category: Category) => `${category.categoryName}`);
+          return { ...book, authors, categories };
+        });
+        setBooks(res);
+      })
+      .catch(error => console.error(error));
+  }, []);
```

Comment:
I'm not sure about this, but would it not be better to write this together in one useEffect? As both will be run on mounting...